### ISSUE TITLE
_gca_resource is not available and causes RuntimeError

### ISSUE THREAD

**Issue by phillies** [googleapis/python-aiplatform#2394]
Trying to run the example from the README creates a RuntimeError and the job is not created. This can be reproduced locally as well as in the cloud shell.
Creating Jobs in Vertex AI via the GUI works.

#### Environment details

  - OS type and version: Windows 10, Ubuntu 22.04 (WSL), Cloud Shell
  - Python version: 3.9 (cloud shell), 3.10 (win+linux), 3.11 (win+linux)
  - pip version: 23.2.1 (windows, linux), 20.3.4 (cloud shell)
  - `google-cloud-aiplatform` version: 1.27.1, 1.28.1

#### Steps to reproduce

  1. Copy the example code from the README
  2. Add import statement and staging bucket
  3. run code
  4. Get RuntimeError because job cannot be created

#### Code example

```python
from google.cloud import aiplatform

job = aiplatform.CustomTrainingJob(
    display_name="my-training-job",
    script_path="training_script.py",
    container_uri="us-docker.pkg.dev/vertex-ai/training/tf-cpu.2-2:latest",
    requirements=["gcsfs==0.7.1"],
    model_serving_container_image_uri="us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-2:latest",
    staging_bucket="gs://test",
)

print(job.display_name)
```

#### Stack trace
```
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    print(job.display_name)
  File "/home/phil/miniconda3/envs/temp/lib/python3.10/site-packages/google/cloud/aiplatform/base.py", line 686, in display_name
    self._assert_gca_resource_is_available()
  File "/home/phil/miniconda3/envs/temp/lib/python3.10/site-packages/google/cloud/aiplatform/base.py", line 1328, in _assert_gca_resource_is_available
    raise RuntimeError(
RuntimeError: CustomTrainingJob resource has not been created.
```

**Comment by sasha-gitg** [comment]
The resource isn't created until `run` is invoked. There is an additional line to execute in the example: https://github.com/googleapis/python-aiplatform/blob/main/README.rst?plain=1#L232-234

**Comment by phillies** [comment]
Ok, but isn't the behaviour then counterintuitive? I provided e.g. `display_name` when creating the job. Throwing an error when I want to read what I just created, e.g. `job.display_name`, is confusing. A hint in the documentation or error message that one *must* call `run()` (or `submit()`) before accessing the object would be helpful.
The same error happens when you execute `job.run(sync=False)` first and then try to read e.g. `display_name`.
